### ISSUE TITLE
EIC plots with shared Y-axis do not have proper Y-axis range 

### ISSUE THREAD

**Issue by wholtz** [biorack/metatlas#476]
example output:
https://drive.google.com/file/d/1JTlDZjkHfPjo59ziivCL1Rir29yiIMiJ/view?usp=sharing

**Comment by wholtz** [comment]
This might only be for C18 or might not be share Y-axis specific. Needs more investigation.

**Comment by wholtz** [comment]
Not C18 or share Y-axis specific -- this is HILIC with independent Y-axis
https://drive.google.com/drive/u/0/folders/1hH1GpYBkMXbfhgKtqG21tJyE5ax8_ROj

**Comment by wholtz** [comment]
Does not occur for:
```
save_compound_eic_pdf(metatlas_dataset, 0, 'out.pdf', sharey=True, max_plots_per_page=1)
```

**Comment by wholtz** [comment]
Minimal reproduction:
``` python
import metatlas.plots.plot_set as plot_set
import matplotlib

class TestPlot(plot_set.Plot):
    def __init__(self, x,  y):
        super().__init__(title='foobar', group_name='grp')
        self.x = x
        self.y = y

    def plot(self, ax: matplotlib.axes.Axes, back_color) -> None:
        """Draw plot of EIC on ax"""
        super().plot(ax)
        ax.plot(self.x, self.y)
        
def save_test_plot_pdf(
    data,
    file_name: str,
    overwrite: bool = False,
    sharey: bool = True,
    max_plots_per_page: int = 30, 
) -> None:
    plots = [TestPlot(x, y) for x, y in data]
    with plot_set.PlotSet(plots, max_plots_per_page, sharey) as plots:
        plots.save_pdf(file_name, title=f"Test Plots", overwrite=overwrite)
        
save_test_plot_pdf([ ( [1,2,3], [1, 3, 1] ), ( [1,2,3], [1, 2, 1] ) ], '/work/test_plot.pdf', overwrite=True, sharey=True)
```

**Comment by wholtz** [comment]
fixed in #518 and #519
